skimage/future/graph/rag.py

```diff
@@ -321,6 +382,100 @@ def rag_mean_color(image, labels, connectivity=2, mode='distance',
     return graph
 
 
+def rag_generic(image, labels, connectivity=2, describe_func=describe_node,
+                weight_func=mean_color, extra_arguments=[],
+                extra_keywords={'sigma':255.0, 'mode':'distance'}):
```

Comment:
Re (4), sure, this is essential! =) In fact, rather than duplicating it in the generic function, reimplement it (but keep the name) by using a call to `region_adjacency_graph`.